Add note on tag requirements (#448)
hashicorp/terraform-docs-common: website/docs/cloud-docs/api-docs/organization-tags.mdx

This API returns the list of tags used with all resources across the organization. Tags can be added to this pool directly or via those resources. Tags deleted here will be removed from all other resources. Tags can be added, applied, removed and deleted in bulk.

> Here are some things to note about tags:

Reviewer: I think my suggestion is still pretty vague, but I can't think of anything better at the moment 😂 :
```suggestion
Tags are subject to the following rules:
```

> - Workspace tags or tags must be one or more characters; can include letters, numbers, colons, hyphens, and underscores; and must begin and end with a letter or number.

Reviewer: We don't use semi-colons outside of code, so splitting this into two bullets:
```suggestion
- Workspace tags or tags must be one or more characters, and can include letters, numbers, colons, hyphens, and underscores. 
- Tags must begin and end with a letter or number, and have a 255 character limit.
```

> - Tags have a 255 character limit.

Reviewer: I think we can nicely combine this into my suggested sentence above since they are both about tag name  restrictions
```suggestion
```

> - Tags must be created for a workspace via the UI or API. Once created, tags can be subsequently assigned to other workspaces in the same organization.

Reviewer: Active voice:
```suggestion
- You can create tags for a workspace using the user interface or the API. After you create a tag, you can assign it to other workspaces in the same organization.
```

> - There is no way to create tags for a workspace via the CLI today.

Reviewer: We only describe the current state of things (so today is assumed):
```suggestion
- You cannot create tags for a workspace using the CLI.
```

> - Tags cannot be set at the Project level today. As a result, there's no inheritance of tags from Projects to Workspaces under a given project.

Reviewer: ```suggestion
- You cannot set tags at the project level, so there is no tag inheritance from projects to workspaces.
```
